common/scripts/audit.js

```diff
@@ -0,0 +1,95 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+const fs = require("fs");
+const path = require("path");
+const { spawn } = require("child_process");
+const { logBuildError, logBuildWarning, throwAfterTimeout } = require("./utils");
+
+const rushCommonDir = path.join(__dirname, "../");
+
+(async () => {
+  const commonTempDir = path.join(rushCommonDir, "config/rush");
+
+  // Npm audit will occasionally take minutes to respond - we believe this is just the npm registry being terrible and slow.
+  // We don't want this to slow down our builds though - we'd rather fail fast and try again later.  So we'll just timeout after 30 seconds.
+  let jsonOut = {};
+  try {
+    console.time("Audit time");
+    jsonOut = await Promise.race([runPnpmAuditAsync(commonTempDir), throwAfterTimeout(180000, "Timed out contacting npm registry.")]);
+    console.timeEnd("Audit time");
+    console.log();
+  } catch (error) {
+    // We want to stop failing the build on transient failures and instead fail only on high/critical vulnerabilities.
+    logBuildWarning(error);
+    process.exit();
+  }
+
+  if (jsonOut.error) {
+    console.error(jsonOut.error.summary);
+    logBuildWarning("Rush audit failed. This may be caused by a problem with the npm audit server.");
+  }
+
+  // A list of temporary advisories excluded from the High and Critical list.
+  // Warning this should only be used as a temporary measure to avoid build failures
+  // for development dependencies only.
+  // All security issues should be addressed asap.
+  const excludedAdvisories = [
+    1674, // https://npmjs.com/advisories/1674
+    1700, // https://npmjs.com/advisories/1700
```

Comment:
@aruniverse are these the ones you're hitting in this pipeline?

Reply:
i copied these directly from the core repo, i assumed id hit the same ones core does for the most part

Reply:
ill double check to be safe